tgui/packages/tgui/interfaces/SpawnPanel/CreateObject.tsx

```diff
@@ -0,0 +1,453 @@
+      value,
+      storageKey: 'spawnpanel-sortBy',
+      setter: setSortBy,
+    });
+  };
+
+  const updateHideMapping = (value: boolean) => {
+    setStateAndStorage({
+      value,
+      storageKey: 'spawnpanel-hideMapping',
+      setter: setHideMapping,
+    });
+  };
+
+  const updateShowIcons = (value: boolean) => {
+    setStateAndStorage({
+      value,
+      storageKey: 'spawnpanel-showIcons',
+      setter: setshowIcons,
+    });
+  };
+
+  const updateShowPreview = (value: boolean) => {
+    setStateAndStorage({
+      value,
+      storageKey: 'spawnpanel-showPreview',
+      setter: setshowPreview,
+    });
+  };
+
+  const sendPreferences = (settings: Partial<SpawnPreferences>) => {
+    const prefsToSend = {
+      hide_icons: showIcons,
+      hide_mappings: hideMapping,
+      sort_by:
+        Object.keys(listTypes).find((key) => listTypes[key] === sortBy) ||
+        'Objects',
+      search_text: searchText,
+      search_by: searchBy ? 'type' : 'name',
+      ...settings,
+    };
+
+    act('create-object-action', prefsToSend);
+  };
+
+  const handleObjectSelect = (obj: string) => {
+    setSelectedObj(obj);
+    act('selected-object-changed', {
+      newObj: obj,
+    });
+    if (currentList[obj]) {
+      props.onIconSettingsChange?.({
+        icon: currentList[obj].icon,
+        iconState: currentList[obj].icon_state,
+      });
+    }
+  };
+
+  return (
+    <Stack vertical fill>
+      <Stack.Item>
+        <Section>
+          <CreateObjectSettings
+            onCreateObject={sendPreferences}
+            setAdvancedSettings={setAdvancedSettings}
+            iconSettings={iconSettings}
+          />
+        </Section>
+      </Stack.Item>
+
+      {showPreview && selectedObj && currentList[selectedObj] && (
+        <Stack.Item>
+          <Section
+            style={{
+              height: '6em',
+            }}
+          >
+            <Stack>
+              <Stack.Item>
+                <Button
+                  width="5em"
+                  height="4.8em"
+                  mb="-3px"
+                  color="transparent"
+                  ml="1px"
+                  style={{
+                    alignContent: 'center',
+                  }}
+                >
+                  <DmIcon
+                    width="4em"
+                    mt="2px"
+                    icon={iconSettings.icon || currentList[selectedObj].icon}
+                    icon_state={
+                      iconSettings.iconState ||
+                      currentList[selectedObj].icon_state
+                    }
+                  />
+                </Button>
+              </Stack.Item>
+              <Stack.Item
+                grow
+                style={{
+                  maxHeight: '4.8em',
+                  overflowY: 'auto',
+                }}
+              >
+                <Stack vertical>
+                  <Stack.Item>
+                    <b>{currentList[selectedObj].name}</b>
```

Comment:
there should be a `bold` prop and `italic` prop on stack.item (for the next instance)